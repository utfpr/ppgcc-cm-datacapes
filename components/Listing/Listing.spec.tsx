import { render } from "@testing-library/react";
import { Listing } from "./index";

const dataTest = [
  {
    id: "aasd34-32-df-sdf3",
    title: "Teste",
    author: "joaozinho",
    email: "teste@teste.com",
  },
  {
    id: "aasd34-32-sddsf-sdf3",
    title: "artigo 2",
    author: "leonardo",
    email: "abc@teste.com",
  },
  {
    id: "aasd34-32-asasf-sdf3",
    title: "Artigo 3",
    author: "oliveira",
    email: "def@gmail.com",
  },
];

describe("Listing components", () => {
  it("renders correctly", () => {
    const { container } = render(<Listing data={dataTest} />);

    expect(container).toMatchSnapshot();
  });

  it("should render a table with the correct number of rows", () => {
    const { container } = render(<Listing data={dataTest} />);

    expect(container.querySelectorAll("tr").length).toBe(dataTest.length + 1);
  });

  it("should render a table with the correct number of columns", () => {
    const { container } = render(<Listing data={dataTest} />);

    expect(container.querySelectorAll("th").length).toBe(dataTest.length);
  });

  it("should not render a table if there is no data", () => {
    const { container } = render(<Listing data={[]} />);

    expect(container.querySelectorAll("tr").length).toBe(1);
  });
});
