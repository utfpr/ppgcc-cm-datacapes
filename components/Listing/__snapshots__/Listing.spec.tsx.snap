// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Listing components renders correctly 1`] = `
<div>
  <div
    class="css-0"
  >
    <div
      class="css-1j0t3zt"
    >
      <div
        class="css-zg32nz"
      >
        <div
          class="css-efd5gz"
        >
          <label
            class="css-1okgzol"
          >
            <input
              class="chakra-input css-c389k8"
              placeholder="Buscar"
            />
            <svg
              class="chakra-icon css-1lpcv4a"
              fill="currentColor"
              focusable="false"
              height="1em"
              stroke="currentColor"
              stroke-width="0"
              viewBox="0 0 24 24"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <g>
                <path
                  d="M0 0h24v24H0z"
                  fill="none"
                />
                <path
                  d="M18.031 16.617l4.283 4.282-1.415 1.415-4.282-4.283A8.96 8.96 0 0 1 11 20c-4.968 0-9-4.032-9-9s4.032-9 9-9 9 4.032 9 9a8.96 8.96 0 0 1-1.969 5.617zm-2.006-.742A6.977 6.977 0 0 0 18 11c0-3.868-3.133-7-7-7-3.868 0-7 3.132-7 7 0 3.867 3.132 7 7 7a6.977 6.977 0 0 0 4.875-1.975l.15-.15z"
                />
              </g>
            </svg>
          </label>
          <a
            class="chakra-button css-5tzbwi"
            href="/users/create"
          >
            <span
              class="chakra-button__icon css-1wh2kri"
            >
              <svg
                aria-hidden="true"
                class="chakra-icon css-1cv0b1u"
                fill="currentColor"
                focusable="false"
                height="1em"
                stroke="currentColor"
                stroke-width="0"
                viewBox="0 0 24 24"
                width="1em"
                xmlns="http://www.w3.org/2000/svg"
              >
                <g>
                  <path
                    d="M0 0h24v24H0z"
                    fill="none"
                  />
                  <path
                    d="M11 11V5h2v6h6v2h-6v6h-2v-6H5v-2z"
                  />
                </g>
              </svg>
            </span>
            Gerar relat√≥rio
          </a>
        </div>
        <table
          class="chakra-table css-0"
          role="table"
        >
          <thead
            class="css-0"
          >
            <tr
              class="css-0"
              role="row"
            >
              <th
                class="css-11p2qph"
              >
                <label
                  class="chakra-checkbox css-1uiwwan"
                >
                  <input
                    class="chakra-checkbox__input"
                    style="border: 0px; clip: rect(0px, 0px, 0px, 0px); height: 1px; width: 1px; margin: -1px; padding: 0px; overflow: hidden; white-space: nowrap; position: absolute;"
                    type="checkbox"
                    value=""
                  />
                  <span
                    aria-hidden="true"
                    class="chakra-checkbox__control css-dnty2r"
                  />
                </label>
              </th>
              <th
                class="css-gnzpcw"
              >
                Titulo
              </th>
              <th
                class="css-w072pj"
              />
            </tr>
          </thead>
          <tbody
            class="css-0"
          >
            <tr
              class="css-0"
              role="row"
            >
              <td
                class="css-zua83t"
                role="gridcell"
              >
                <label
                  class="chakra-checkbox css-1uiwwan"
                >
                  <input
                    class="chakra-checkbox__input"
                    style="border: 0px; clip: rect(0px, 0px, 0px, 0px); height: 1px; width: 1px; margin: -1px; padding: 0px; overflow: hidden; white-space: nowrap; position: absolute;"
                    type="checkbox"
                    value=""
                  />
                  <span
                    aria-hidden="true"
                    class="chakra-checkbox__control css-dnty2r"
                  />
                </label>
              </td>
              <td
                class="css-40pc19"
                role="gridcell"
              >
                Teste
              </td>
              <td
                class="css-0"
                role="gridcell"
              >
                <a
                  class="chakra-button css-1nbjkjt"
                >
                  <span
                    class="chakra-button__icon css-1wh2kri"
                  >
                    <svg
                      aria-hidden="true"
                      class="chakra-icon css-ky7im0"
                      fill="currentColor"
                      focusable="false"
                      height="1em"
                      stroke="currentColor"
                      stroke-width="0"
                      viewBox="0 0 24 24"
                      width="1em"
                      xmlns="http://www.w3.org/2000/svg"
                    >
                      <g>
                        <path
                          d="M0 0h24v24H0z"
                          fill="none"
                        />
                        <path
                          d="M7 6V3a1 1 0 0 1 1-1h12a1 1 0 0 1 1 1v14a1 1 0 0 1-1 1h-3v3c0 .552-.45 1-1.007 1H4.007A1.001 1.001 0 0 1 3 21l.003-14c0-.552.45-1 1.007-1H7zm2 0h8v10h2V4H9v2zm-2 5v2h6v-2H7zm0 4v2h6v-2H7z"
                        />
                      </g>
                    </svg>
                  </span>
                  
                </a>
              </td>
            </tr>
            <tr
              class="css-0"
              role="row"
            >
              <td
                class="css-zua83t"
                role="gridcell"
              >
                <label
                  class="chakra-checkbox css-1uiwwan"
                >
                  <input
                    class="chakra-checkbox__input"
                    style="border: 0px; clip: rect(0px, 0px, 0px, 0px); height: 1px; width: 1px; margin: -1px; padding: 0px; overflow: hidden; white-space: nowrap; position: absolute;"
                    type="checkbox"
                    value=""
                  />
                  <span
                    aria-hidden="true"
                    class="chakra-checkbox__control css-dnty2r"
                  />
                </label>
              </td>
              <td
                class="css-40pc19"
                role="gridcell"
              >
                artigo 2
              </td>
              <td
                class="css-0"
                role="gridcell"
              >
                <a
                  class="chakra-button css-1nbjkjt"
                >
                  <span
                    class="chakra-button__icon css-1wh2kri"
                  >
                    <svg
                      aria-hidden="true"
                      class="chakra-icon css-ky7im0"
                      fill="currentColor"
                      focusable="false"
                      height="1em"
                      stroke="currentColor"
                      stroke-width="0"
                      viewBox="0 0 24 24"
                      width="1em"
                      xmlns="http://www.w3.org/2000/svg"
                    >
                      <g>
                        <path
                          d="M0 0h24v24H0z"
                          fill="none"
                        />
                        <path
                          d="M7 6V3a1 1 0 0 1 1-1h12a1 1 0 0 1 1 1v14a1 1 0 0 1-1 1h-3v3c0 .552-.45 1-1.007 1H4.007A1.001 1.001 0 0 1 3 21l.003-14c0-.552.45-1 1.007-1H7zm2 0h8v10h2V4H9v2zm-2 5v2h6v-2H7zm0 4v2h6v-2H7z"
                        />
                      </g>
                    </svg>
                  </span>
                  
                </a>
              </td>
            </tr>
            <tr
              class="css-0"
              role="row"
            >
              <td
                class="css-zua83t"
                role="gridcell"
              >
                <label
                  class="chakra-checkbox css-1uiwwan"
                >
                  <input
                    class="chakra-checkbox__input"
                    style="border: 0px; clip: rect(0px, 0px, 0px, 0px); height: 1px; width: 1px; margin: -1px; padding: 0px; overflow: hidden; white-space: nowrap; position: absolute;"
                    type="checkbox"
                    value=""
                  />
                  <span
                    aria-hidden="true"
                    class="chakra-checkbox__control css-dnty2r"
                  />
                </label>
              </td>
              <td
                class="css-40pc19"
                role="gridcell"
              >
                Artigo 3
              </td>
              <td
                class="css-0"
                role="gridcell"
              >
                <a
                  class="chakra-button css-1nbjkjt"
                >
                  <span
                    class="chakra-button__icon css-1wh2kri"
                  >
                    <svg
                      aria-hidden="true"
                      class="chakra-icon css-ky7im0"
                      fill="currentColor"
                      focusable="false"
                      height="1em"
                      stroke="currentColor"
                      stroke-width="0"
                      viewBox="0 0 24 24"
                      width="1em"
                      xmlns="http://www.w3.org/2000/svg"
                    >
                      <g>
                        <path
                          d="M0 0h24v24H0z"
                          fill="none"
                        />
                        <path
                          d="M7 6V3a1 1 0 0 1 1-1h12a1 1 0 0 1 1 1v14a1 1 0 0 1-1 1h-3v3c0 .552-.45 1-1.007 1H4.007A1.001 1.001 0 0 1 3 21l.003-14c0-.552.45-1 1.007-1H7zm2 0h8v10h2V4H9v2zm-2 5v2h6v-2H7zm0 4v2h6v-2H7z"
                        />
                      </g>
                    </svg>
                  </span>
                  
                </a>
              </td>
            </tr>
          </tbody>
        </table>
        <div
          class="chakra-stack css-1n17tlq"
        >
          <div
            class="css-kbn7if"
          >
            <strong>
              0
            </strong>
             - 
            <strong>
              10
            </strong>
             de 
            <strong>
              100
            </strong>
          </div>
          <div
            class="chakra-stack css-1689b8b"
          >
            <button
              class="chakra-button css-1jn0x3k"
              disabled=""
              type="button"
            >
              1
            </button>
            <button
              class="chakra-button css-1ysjku8"
              type="button"
            >
              2
            </button>
            <p
              class="chakra-text css-8t3u5b"
            >
              ...
            </p>
            <button
              class="chakra-button css-1ysjku8"
              type="button"
            >
              10
            </button>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
